src/renderer/components/version-select.tsx

```diff
@@ -93,18 +88,16 @@ export function getItemLabel({ source, state, name }: RunnableVersion): string {
  * @param {RunnableVersion} { state }
  * @returns
  */
-export function getItemIcon({ state }: RunnableVersion) {
-  switch (state) {
-    case InstallState.missing:
-      return 'cloud';
-    case InstallState.installing:
-      return 'compressed';
-    case InstallState.installed:
-    case InstallState.downloaded:
-      return 'saved';
-    case InstallState.downloading:
-      return 'cloud-download';
-  }
+export function getItemIcon({ state }: RunnableVersion): IconName {
+  const installStateIcons: Record<InstallState, IconName> = {
+    missing: 'cloud',
+    downloading: 'cloud-download',
+    downloaded: 'compressed',
+    installing: 'compressed',
+    installed: 'saved',
+  } as const;
+
+  return installStateIcons[state] || '';
```

Comment:
Note for posterity (and my own future reference). I was concerned this change might remove the exhaustiveness checking you get in TypeScript with a switch statement with no `default`, but it looks like TypeScript also does exhaustiveness checking with a `Record` when the key is a finite type. :+1: